### DISCUSSION TITLE
traversal nodes, destroying the entire network

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [libp2p/rust-libp2p#3500] [Q&A] [answered]
Point-to-point, as long as there is a way to traversal nodes, there is a way to destroy the entire network
P2P opponent is the state: the government ordered ISP to traversal the nodes,
How can libp2p solve this problem ?

**Comment by thomaseizinger** [comment, marked as answer]
Can you elaborate a bit more on what you are trying to achieve? At its heart libp2p offers you a set of modules for building peer to peer services. That includes e.g. authentication and the ability to open light-weight multiplexed streams in either direction.

libp2p has practically no opinion on what you do with these capabilities. Whether your application is censorship-resistant or can withstand eclipse attacks or other things depends pretty much entirely on the protocols you use and how you design those.

For example, if you build your application with only a central bootnode but otherwise no capability to connect to other nodes, your network can easily be taken down by attacking that one.
